modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,366 @@
+                            case DAYS:
+                                if (element.getValue().equalsIgnoreCase("all")) {
+                                    days = -1;
+                                } else {
+                                    days = timeToLong(element.getValue());
+                                }
+                                retentionPeriods.put(TimePeriod.Duration.DAYS, days);
+                                break;
+                            case MONTHS:
+                                if (element.getValue().equalsIgnoreCase("all")) {
+                                    months = -1;
+                                } else {
+                                    months = timeToLong(element.getValue());
+                                }
+                                retentionPeriods.put(TimePeriod.Duration.MONTHS, months);
+                                break;
+                            case YEARS:
+                                if (element.getValue().equalsIgnoreCase("all")) {
+                                    years = -1;
+                                } else {
+                                    years = timeToLong(element.getValue());
+                                }
+                                retentionPeriods.put(TimePeriod.Duration.YEARS, years);
+                                break;
+                            case HOURS:
+                                if (element.getValue().equalsIgnoreCase("all")) {
+                                    hours = -1;
+                                } else {
+                                    hours = timeToLong(element.getValue());
+                                }
+                                retentionPeriods.put(TimePeriod.Duration.HOURS, hours);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private Long getPurgeExecutionInterval() {
+        return purgeExecutionInterval;
+    }
+
+    public boolean isPurgingEnabled() {
+        return purgingEnabled;
+    }
+
+    @Override
+    public void run() {
+        long currentTime = System.currentTimeMillis();
+        long purgeTime;
+        Object[] purgeTimes = new Object[1];
+        List<VariableExpressionExecutor> variableExpressionExecutorList = new ArrayList<>();
+        Attribute attribute;
+        VariableExpressionExecutor variableExpressionExecutor;
+        ComplexEventChunk<StateEvent> eventChunk = new ComplexEventChunk<>(true);
+        BaseIncrementalDataPurgingValueStore baseIncrementalDataPurgingValueStore = new
+                BaseIncrementalDataPurgingValueStore(currentTime, streamEventPool);
+        attribute = new Attribute(INTERNAL_AGG_TIMESTAMP_FIELD, Attribute.Type.LONG);
+        variableExpressionExecutor = new VariableExpressionExecutor(attribute, 0, 1);
+        variableExpressionExecutorList.add(variableExpressionExecutor);
```

Comment:
Move all these initialization logic out of run method and do it only once in the init()